snuba/cli/query_comparer.py

```diff
@@ -139,86 +53,186 @@ def query_comparer(
     setup_logging(log_level)
     setup_sentry()
 
-    base = open(base_file)
-    upgrade = open(upgrade_file)
+    uploader = GCSUploader(gcs_bucket)
+    file_manager = FileManager(uploader)
+    blob_getter = BlobGetter(uploader)
 
-    base_reader = csv.reader(base)
-    upgrade_reader = csv.reader(upgrade)
+    blob_prefixes = blob_getter._get_sub_dir_prefixes(prefix="")
+    result_prefixes = sorted([p for p in blob_prefixes if p.startswith("results")])
 
-    def query_measurement(row: Any) -> QueryMeasurement:
-        return QueryMeasurement(
-            query_id=row[0],
-            query_duration_ms=int(row[1]),
-            result_rows=int(row[2]),
-            result_bytes=int(row[3]),
-            read_rows=int(row[4]),
-            read_bytes=int(row[5]),
-        )
+    if len(result_prefixes) != 2:
+        logger.info("Not enough results to compare.")
+        return
 
-    mismatches = []
-    total_rows = 0
-    for v1_row, v2_row in itertools.zip_longest(base_reader, upgrade_reader):
-        if v1_row[0] == "query_id":
-            # csv header row
+    def get_matched_pairs() -> Sequence[Tuple[str, str]]:
```

Comment:
Can you document what this function does and what it returns. Similar to what you have in the PR description.